libnetwork/drivers/bridge/bridge_linux.go

```diff
@@ -428,6 +435,15 @@ func (n *bridgeNetwork) getNATDisabled() (ipv4, ipv6 bool) {
 	return n.config.GwModeIPv4.routed(), n.config.GwModeIPv6.routed()
 }
 
+func (n *bridgeNetwork) isGwModeUnprotected(v iptables.IPVersion) bool {
+	n.Lock()
+	defer n.Unlock()
```

Comment:
Mostly curious; we have so much locking everywhere; do we have good insight what exactly is it synchronising? (was contemplating if it's purely `n.config`, and if we should have an accessor for that to return it 🤔